### DISCUSSION TITLE
Mapping custom calibre metadata to koreaders's tag search/browse?

### DISCUSSION THREAD

**Discussion by procyon108** [koreader/koreader#11334] [General]
I'd been using Koreader on my Kobo devices for years for PDFs. I'm now fully switched over for all my ebooks. My books have 3-8 tags each. To sort them on Kobo, I use very broad 'subject' metadata (via Calibre custom columns). Nickle handles this with the Caliber Kobo plugin/driver. I need to manage these collections in Koreader's FM now. I tried using a custom template for sending the files, but the paths get randomly truncated and the results are unpredictable.

I understand why enabling arbitrary user metadata in Koreader's search is not a good idea. I'm a lower-level Calibre power-user and I know how extreme it can get. I'd like to request a compromise. I don't use tags on the e-ereader. But I do use them, so I don't want to replace them with my "subject" metadata in my Calibre library as had been suggested to me, but It would be great if I could have koreader's (unused) tag lookup mapped to my subject.

I'm open to any solution to finding my books better. (on my device). My next move is to try calibre2opds in Trook mode. Koreader is becoming a favorite application of mine, and I'm not giving up on this.

**Comment by NiLuJe** [comment]
Might be as simple as something along those lines:

```diff
diff --git a/plugins/calibre.koplugin/metadata.lua b/plugins/calibre.koplugin/metadata.lua
index fa78d21ab..a33119639 100644
--- a/plugins/calibre.koplugin/metadata.lua
+++ b/plugins/calibre.koplugin/metadata.lua
@@ -23,6 +23,7 @@ local used_metadata = {
     "title",
     "authors",
     "tags",
+    "whatever",
     "series",
     "series_index"
 }
@@ -34,6 +35,7 @@ local search_used_metadata = {
     "title",
     "authors",
     "tags",
+    "whatever",
     "series",
     "series_index"
 }
diff --git a/plugins/calibre.koplugin/search.lua b/plugins/calibre.koplugin/search.lua
index 432ffaa11..cdc6a2330 100644
--- a/plugins/calibre.koplugin/search.lua
+++ b/plugins/calibre.koplugin/search.lua
@@ -209,7 +209,7 @@ function CalibreSearch:ShowSearch()
                     enabled = true,
                     callback = function()
                         self.search_value = self.search_dialog:getInputText()
-                        self.lastsearch = "tags"
+                        self.lastsearch = "whatever"
                         self:close()
                     end,
                 },
```

(You may need to scrap our metadata cache, or touch your calibre on-device cache to request a reparse).

**Reply by tarkeez** [reply]
To do this on android, would I need to either repackage the apk or have access to the system files?

Is it possible to do it with a user plugin?

**Reply by pazos** [reply]
copy the calibre plugin with another name, change metadata, disable original plugin, make changes on your own and enable your particular plugin from menus. That's all :)
